File: task_equivalents/llm_translations/llama/optimized/r/data_manipulation_preprocessing/02_merge_aggregate_reshape.R
library(readr)
library(dplyr)

College <- read_csv('data/csv/College.csv')
College <- College %>% 
  column_to_rownames(var = names(College)[1]) %>% 
  mutate(Private01 = ifelse(Private == 'Yes', 1, 0)) %>% 
  select(-Private, -Apps)

set.seed(1)
train_index <- sample(nrow(College), size = nrow(College) / 2)
xtrain <- College[train_index, setdiff(names(College), 'Apps')] 
ytrain <- College[train_index, 'Apps']
xtest <- College[-train_index, setdiff(names(College), 'Apps')] 
ytest <- College[-train_index, 'Apps']

File: task_equivalents/llm_translations/llama/optimized/python/data_manipulation_preprocessing/02_merge_aggregate_reshape.py
import pandas as pd
import numpy as np

# Load the data
college = pd.read_csv('data/csv/College.csv', index_col=0)

# Create a new column 'Private01' and drop 'Private'
college['Private01'] = (college['Private'] == 'Yes').astype(int)
college = college.drop('Private', axis=1)

# Split data into X and Y
X = college.drop('Apps', axis=1).apply(pd.to_numeric)
Y = college['Apps']

# Split data into training and test sets
np.random.seed(1)
idx = np.random.permutation(len(college))
train_idx, test_idx = idx[:len(college)//2], idx[len(college)//2:]
xtrain, xtest = X.iloc[train_idx], X.iloc[test_idx]
ytrain, ytest = Y.iloc[train_idx], Y.iloc[test_idx]